### PR TITLE
Checkbox test

### PR DESCRIPTION
**Pull request by MarcoSantarelli**
**Related tasks:**
was added:
* checkbox test

**Related cards:**
* FID-33

**Before testing:**
1. Clone the PR 
2. run the angular project
3. run the command "ng test"
---

### CODE REVIEW

File: src/app/atoms/check-box/che-checkbox.component.spec.ts [hunk 1]
```diff
@@ -1,26 +1,311 @@
...
+            // Assert
+            expect(checkbox.nativeElement.getAttribute('label')).toEqual('this is a label');
+
+        });
+
+
+    });
+
+
+    describe('If it is disabled', () => {
```
Comment (danabalon) [inline src/app/atoms/check-box/che-checkbox.component.spec.ts:135]: if disabled should to be not clicked? (other test)
and not focusable? (other test)

File: src/app/atoms/check-box/che-checkbox.component.spec.ts [hunk 2]
```diff
@@ -1,26 +1,311 @@
...
+
+            // Assert
+            expect(checkbox.nativeElement.getAttribute('disabled')).toBeTruthy('Should have attribute attribute disabled');
+
+        });
+
+
+    });
+
+    describe('If it is unchecked and clicked', () => {
```
Comment (danabalon) [inline src/app/atoms/check-box/che-checkbox.component.spec.ts]: Only "If it is unchecked". The action should to be in the test description.

File: src/app/atoms/check-box/che-checkbox.component.spec.ts [hunk 3]
```diff
@@ -1,26 +1,311 @@
...
+            // Arrange
+
+            fixture = createComponent(CheCheckboxF4Component);
+            component = fixture.debugElement.componentInstance;
+            fixture.detectChanges();
+
+        });
+
+
+        it('should change from unchecked to marked and transparent bg to bg color', () => {
```
Comment (danabalon) [inline src/app/atoms/check-box/che-checkbox.component.spec.ts]: Should to change to checked and change from transparent bg to bg color to click.

File: src/app/atoms/check-box/che-checkbox.component.spec.ts [hunk 4]
```diff
@@ -1,26 +1,311 @@
...
+            fixture.detectChanges();
+            // Assert
+            expect(checkbox.nativeElement.classList.contains('che-checkbox-checked')).toBeTruthy('Should have class "che-checkbox-checked"');
+
+        });
+
+
+    });
+
+    describe('If it is checked and clicked', () => {
```
Comment (danabalon) [inline src/app/atoms/check-box/che-checkbox.component.spec.ts]: If it is checked

File: src/app/atoms/check-box/che-checkbox.component.spec.ts [hunk 5]
```diff
@@ -1,26 +1,311 @@
...
+
+        beforeEach(() => {
+            // Arrange
+            fixture = createComponent(CheCheckboxF4Component);
+            component = fixture.componentInstance;
+            fixture.detectChanges();
+        });
+
+
+        it('should change from checked to unchecked and  bg color to transparent bg', () => {
```
Comment (danabalon) [inline src/app/atoms/check-box/che-checkbox.component.spec.ts]: Should to change to unchecked and change from bg color to transparent bg to click.

File: src/app/atoms/check-box/che-checkbox.component.spec.ts [hunk 6]
```diff
@@ -1,26 +1,311 @@
...
+            // Assert
+            expect(checkbox.nativeElement.classList.contains('che-checkbox-checked')).toBeFalsy('Should have not class "che-checkbox-checked"');
+
+        });
+
+
+    });
+
+
+    describe('If it is focused', () => {
```
Comment (danabalon) [inline src/app/atoms/check-box/che-checkbox.component.spec.ts]: move to others describes... See the document "Test Cases Definition"

File: src/app/atoms/check-box/che-checkbox.component.spec.ts [hunk 7]
```diff
@@ -1,26 +1,311 @@
...
+            fixture.detectChanges();
+            // Assert
+            expect(checkbox.nativeElement.classList.contains('che-checkbox-focused')).toBeTruthy('Should have class "che-checkbox-focused"');
+        });
+
+
+    });
+
+
+    describe('If it is mouseout', () => {
```
Comment (danabalon) [inline src/app/atoms/check-box/che-checkbox.component.spec.ts]: move to others describes... See the document "Test Cases Definition"

File: src/app/atoms/input/che-input.component.spec.ts [hunk 8]
```diff
@@ -1,25 +1,315 @@
...
+            //   fixture.detectChanges();
+            //
+            //   console.log(input.nativeElement.classList);
+            //
+            //     // Assert
+            //   expect(input.attributes.value).toBe('something');
+            //
+            //
+            //
+            // });
```
Comment (danabalon) [inline src/app/atoms/input/che-input.component.spec.ts:88]: commented code

File: src/app/atoms/input/che-input.component.spec.ts [hunk 9]
```diff
@@ -1,25 +1,315 @@
...
+    //     // checking the next estate
+    //
+    //     expect(input.nativeElement.value)
+    //       .toBe('a value', 'should to have the text');
+    //
+    //     expect(divGroup.nativeElement.classList.contains('has-danger'))
+    //       .toBe(false, 'should not to have the class has-danger');
+    //
+    //   });
+    // });
```
Comment (danabalon) [inline src/app/atoms/input/che-input.component.spec.ts:180]: commented code

File: src/app/demo/checkbox/checkbox.component.html [hunk 10]
```diff
@@ -5,7 +5,7 @@ <h3>Checkbox </h3>
 
 <div class="row">
     <div class="col-12">
-        <che-checkbox [(ngModel)]='checkbox' color="primary"  label='this is a checkbox' ></che-checkbox> <br>
+        <che-checkbox [(ngModel)]='checkbox' color="danger"   label='this is a checkbox' ></che-checkbox> <br>
```
Comment (danabalon) [inline src/app/demo/checkbox/checkbox.component.html]: should be the primary color

**Review by danabalon — APPROVED** [review]
Approved with some changes

File: karma.conf.js [hunk 11]
```diff
@@ -5,6 +5,10 @@ module.exports = function (config) {
   config.set({
     basePath: '',
     frameworks: ['jasmine', '@angular-devkit/build-angular'],
+    files:[
+        "node_modules/@angular/material/prebuilt-themes/indigo-pink.css",
+        // "src/styles.scss",
```
Comment (aderenzis) [inline karma.conf.js:10]: commented line

**Review by aderenzis — APPROVED** [review]
Approved with a few English improvements

**Pull request merged**